feature-libs/cart/saved-cart/components/details/saved-cart-detail-items/saved-cart-detail-items.component.spec.ts

```diff
@@ -1,52 +1,177 @@
+  takeUserId(_loggedIn: boolean): Observable<string> {
+    return of(mockUserId);
+  }
+}
+
+class MockRoutingService implements Partial<RoutingService> {
+  go(): void {}
+}
+class MockGlobalMessageService implements Partial<GlobalMessageService> {
+  add(
+    _text: string | Translatable,
+    _type: GlobalMessageType,
+    _timeout?: number
+  ): void {}
 }
 
 describe('SavedCartDetailItemsComponent', () => {
   let component: SavedCartDetailItemsComponent;
   let fixture: ComponentFixture<SavedCartDetailItemsComponent>;
+  let savedCartService: SavedCartService;
+  let globalMessageService: GlobalMessageService;
+  let routingService: RoutingService;
 
-  beforeEach(() => {
-    TestBed.configureTestingModule({
-      imports: [StoreModule.forRoot({})],
-      declarations: [SavedCartDetailItemsComponent],
-      providers: [
-        {
-          provide: SavedCartService,
-          useClass: MockSavedCartService,
-        },
-        {
-          provide: SavedCartDetailService,
-          useClass: MockSavedCartDetailService,
-        },
-      ],
-    }).compileComponents();
-
-    fixture = TestBed.createComponent(SavedCartDetailItemsComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
+  beforeEach(
+    waitForAsync(() => {
+      TestBed.configureTestingModule({
+        imports: [StoreModule.forRoot({}), I18nTestingModule],
+        declarations: [
+          SavedCartDetailItemsComponent,
+          MockCartItemListComponent,
+        ],
+        providers: [
+          {
+            provide: SavedCartService,
+            useClass: MockSavedCartService,
+          },
+          {
+            provide: SavedCartDetailService,
+            useClass: MockSavedCartDetailService,
+          },
+          {
+            provide: UserIdService,
+            useClass: MockUserIdService,
+          },
+          { provide: RoutingService, useClass: MockRoutingService },
+          { provide: GlobalMessageService, useClass: MockGlobalMessageService },
+        ],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(SavedCartDetailItemsComponent);
+      component = fixture.componentInstance;
+
+      savedCartService = TestBed.inject(SavedCartService);
+      globalMessageService = TestBed.inject(GlobalMessageService);
+      routingService = TestBed.inject(RoutingService);
+
+      spyOn(routingService, 'go').and.callThrough();
+      spyOn(globalMessageService, 'add').and.callThrough();
+      spyOn(savedCartService, 'deleteSavedCart').and.callThrough();
+
+      cart$.next(mockSavedCart);
+
+      fixture.detectChanges();
+    })
+  );
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should NOT delete saved cart when there are cart entries', () => {
+    component.savedCart$
+      .subscribe((savedCart) => {
+        expect(savedCart).toEqual(mockSavedCart);
+        expect(routingService.go).not.toHaveBeenCalledWith({
```

Comment:
No point in testing here `not.toHaveBeenCalledWith`. You want to make sure that you don't call these methods at all with `not.toHaveBennCalled`